src/sc/sidechain.cpp

```diff
@@ -713,17 +713,30 @@ int CSidechain::getNumBlocksForScFeeCheck()
 
 int CSidechain::getMaxSizeOfScFeesContainers()
 {
-    int epochLength = fixedParams.withdrawalEpochLength;
-    int numBlocks = getNumBlocksForScFeeCheck();
-
-    //assert(epochLength > 0);
-    if (epochLength == 0) {
-        maxSizeOfScFeesContainers = 1;
-    } else {
-        maxSizeOfScFeesContainers = numBlocks / epochLength;
-        if (maxSizeOfScFeesContainers == 0 || (numBlocks % epochLength != 0) )
-        {
-            maxSizeOfScFeesContainers++;
+    if (maxSizeOfScFeesContainers == -1) {
+        maxSizeOfScFeesContainers = getNumBlocksForScFeeCheck();
+
+        if (!isNonCeasing()) {
+            int epochLength = fixedParams.withdrawalEpochLength;
+            int numBlocks = getNumBlocksForScFeeCheck();
```

Comment:
Shouldn't `epochLength` and `numBlocks` be `const` ?
Also wouldn't be better to instantiate `numBlocks` in the `else` branch ? (it has no use in the `if` branch